app/cdap/components/SourceControlManagement/OperationAlert.tsx

```diff
@@ -0,0 +1,147 @@
+ * the License.
+ */
+
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import T from 'i18n-react';
+import { IOperationRun } from './types';
+import Alert from '@material-ui/lab/Alert';
+import AlertTitle from '@material-ui/lab/AlertTitle';
+import {
+  getOperationRunMessage,
+  getOperationRunTime,
+  getOperationStartTime,
+  getOperationStatusType,
+  parseOperationResource,
+} from './helpers';
+import { Button, CircularProgress } from '@material-ui/core';
+import { stopOperation } from './store/ActionCreator';
+import { getCurrentNamespace } from 'services/NamespaceStore';
+import { OperationStatus } from './OperationStatus';
+import ExpandLess from '@material-ui/icons/ExpandLess';
+import ExpandMore from '@material-ui/icons/ExpandMore';
+import { AlertErrorView } from './styles';
+
+interface IOperationBannerProps {
+  operation: IOperationRun;
+}
+
+const StyledDiv = styled.div`
+  margin-bottom: 24px;
+`;
+
+const ExpandWrapper = styled.div`
+  height: 100%;
+  padding-top: 12px;
+`;
+
+const PREFIX = 'features.SourceControlManagement';
+
+export const OperationAlert = ({ operation }: IOperationBannerProps) => {
+  const [viewErrorExpanded, setViewErrorExpanded] = useState(false);
+
+  const getOperationAction = () => {
+    if (
+      operation.status === OperationStatus.STARTING ||
+      operation.status === OperationStatus.RUNNING
+    ) {
+      return (
+        <Button
+          color="inherit"
+          size="small"
+          onClick={stopOperation(getCurrentNamespace(), operation)}
+        >
+          {T.translate(`${PREFIX}.stopOperation`)}
+        </Button>
+      );
+    }
+
+    if (operation.status === OperationStatus.STOPPING) {
+      return (
+        <Button color="inherit" size="small" disabled={true} startIcon={<CircularProgress />}>
+          {T.translate(`${PREFIX}.stoppingOperation`)}
+        </Button>
+      );
+    }
+
+    if (operation.status === OperationStatus.FAILED) {
+      return (
+        <ExpandWrapper>
+          <Button
+            color="inherit"
+            size="small"
+            onClick={() => setViewErrorExpanded((isExpanded) => !isExpanded)}
+          >
+            {viewErrorExpanded ? <ExpandLess /> : <ExpandMore />}
+          </Button>
+        </ExpandWrapper>
+      );
+    }
+
+    return undefined;
+  };
+
+  const renderOperationTime = () => {
+    const startedPrefix = T.translate(`${PREFIX}.operationStartedAt`).toString();
+    const startTime = getOperationStartTime(operation);
+    const timeTaken = getOperationRunTime(operation);
+    if (!timeTaken) {
+      return `${startedPrefix} ${startTime}`;
+    }
+
+    const timeTakenPrefix = operation.done
+      ? T.translate(`${PREFIX}.operationRanFor`).toString()
+      : T.translate(`${PREFIX}.operationRunningFor`).toString();
+
+    return `${startedPrefix} ${startTime}, ${timeTakenPrefix} ${timeTaken}`;
```

Comment:
We should not be using template literals to generate localized strings - some languages will expect segments to be in a different order. To be fully internationalized, this should be done through `T.translate` (L102 also)

Reply:
Agreed. This current implementation may not work properly in all cases, especially in RTL languages. Will update it.

Reply:
Updated.